index.js

```diff
@@ -706,7 +707,8 @@ if (require.main !== module && check_dependencies()) {
 
           if (err instanceof Error) {
             if (err instanceof EthicalAdsWarning && !is_debug) {
-              // Skip reporting these warnings for now, unless debugging.
+              // Report these at a lower log level
+              console.debug(err.message);
```

Comment:
I think in the past people have not wanted even debug messages in the console. Can we make this configurable with a `data-ea-log=false` parameter?

Reply:
I wouldn't add a data attribute for that, seems this logic should actually be altered a bit

Reply:
`is_debug` comes from `(global|window).debug` variable, which shouldn't normally be used. This is would only console debug log while `window.debug == true`, and will skip the following console error log in all cases of a warning error:

```js
            if (err instanceof EthicalAdsWarning) {
              if (is_debug) {
                // Report these at a lower log level
                console.debug(err.message);
              }
              return;
            }
```

Reply:
As written, this will print to the console (at debug -- which is off by default on Chrome) for these warnings. There are only 2 warnings currently which are "the ad is blocked" and "there were no ads to show". I think these are worth printing at debug since they shouldn't happen frequently and when they do people might be curious why.

Reply:
After some internal discussion, I think we're going to go with the logic as written. However, if people are unhappy with debug logs we can change it to:

- if error is a warning and debug = true: log via console.debug (currently, debug shouldn't be true but if it is, we log warnings as errors)
- if error is a warning and debug = false: skip logging entirely (this is the normal case and we'll log at debug currently)